### ISSUE TITLE
Can @psalm-assert support Union types?

### ISSUE THREAD

**Issue by jsiefer** [vimeo/psalm#4414]
Hi,

Can @psalm-assert support Union types? - Or allow multiple @psalm-assert tags?

https://psalm.dev/r/c0ce981697

**Comment by muglug** [comment]
This works for `@psalm-assert`: https://psalm.dev/r/68c1e782f4
But doesn't for `@psalm-assert-if-true`: https://psalm.dev/r/cc9b489f62

**Comment by egst** [comment]
It seems like `@psalm-assert-if-true A|B $x` still doesn't work. Or am I missing something?

The only way I've managed to make an assertion that does `$x is A || $x is B` is by negating a result of a function that actually does `$x is not A && $x is not B` that uses two anotations: `@psalm-assert-if-true !A $x` and `@psalm-assert-if-true !B $x`. It technically works, but reading code like `!isNotFooOrBar($x)` is really confusing compared to just `isFooOrBar($x)` instead.

Here are all the variations that I've tried: https://psalm.dev/r/1fa5692e55
